resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/internal/ClientResponse.java

```diff
@@ -401,7 +408,7 @@ public boolean bufferEntity()
    {
       abortIfClosed();
       if (bufferedEntity != null) return true;
-      if (entity != null) return false;
+      if (streamRead) return false;
       if (metadata.getFirst(HttpHeaderNames.CONTENT_TYPE) == null) return false;
       InputStream is = getInputStream();

```

Comment:
Based on @asoldano 's comment here https://github.com/resteasy/Resteasy/pull/1244#discussion_r133709884, don't you think guys that we also have to  make `bufferEntity()` method `synchronized` since it also access and read the stream and concurrency can occurs with `finalize()` method that close this stream?

Reply:
mmh... I would leave the ProcessingException there, the javadoc says that it's meant to be thrown if there was an error while buffering the stream (it's a RuntimeException, so it's not declared)